roles/reproducer/tasks/ci_data.yml

```diff
@@ -52,13 +52,17 @@
             zuul_inventory.all.hosts.keys() |
             select('match', '^compute.*') | length
            }}
+        zuul_params: >-
+          {{
+            lookup('file', _reproducer_basedir ~ '/parameters/zuul-params.yml') |
+            from_yaml
+          }}
       ansible.builtin.set_fact:
         cacheable: true
         ci_job_networking: "{{ zuul_inventory.all.hosts.crc.crc_ci_bootstrap_networking }}"
         use_content_provider: >-
           {{
-            zuul_inventory.all.hosts.crc.content_provider_registry_ip is defined and
-            zuul_inventory.all.hosts.crc.cifmw_operator_build_output is defined
+            zuul_params.content_provider_registry_ip is defined
```

Comment:
looks good. we only  set that when content provider runs  https://github.com/openstack-k8s-operators/ci-framework/blob/5bae6d325fee17999f93d4d4dfb0c4873febcd11/ci/playbooks/content_provider/run.yml#L50

only unknown to me is where/how you are grabbing the parameters/zuul-params.yml but I will assume that the content provider zuul vars are included in that

Reply:
@marios for reference the zuul params are fetched in the `ci_data.yml` task file: https://github.com/openstack-k8s-operators/ci-framework/blob/5bae6d325fee17999f93d4d4dfb0c4873febcd11/roles/reproducer/tasks/ci_data.yml#L23

Reply:
indeed, we fetch earlier selected CI artifacts in order to know how many computes are needed, among things. We're then just parsing the zuul parameters and are able to find out if we have a content-provider or not.